test/core/base/rule.js

```diff
@@ -1662,144 +1662,209 @@ describe('Rule', function() {
         assert.lengthOf(result.nodes, 1);
       });
     });
+
+    it('should mark checks as incomplete if reviewOnFail is set to true', function () {
```

Comment:
I think you should test each type separately. This test can pass if two of the three types are broken.